src/math.ts

```diff
@@ -162,48 +183,59 @@ export const isTokenX = async (candidate: string, compareTo: string): Promise<bo
   ).returns
 }
 
+export async function getTickAtSqrtPrice(sqrtPrice: bigint, tickSpacing: bigint): Promise<bigint> {
+  return (
+    await CLAMM.tests.getTickAtSqrtPrice({
+      testArgs: { sqrtPrice: { v: sqrtPrice }, tickSpacing }
+    })
+  ).returns
+}
+
 export const calculateFee = async (
   pool: Pool,
   position: Position,
   lowerTick: Tick,
   upperTick: Tick
 ): Promise<[bigint, bigint]> => {
-  return (
+  const returns = (
     await Utils.tests.calculateFee({
       testArgs: {
         tickLowerIndex: lowerTick.index,
-        tickLowerFeeGrowthOutsideX: lowerTick.feeGrowthOutsideX,
-        tickLowerFeeGrowthOutsideY: lowerTick.feeGrowthOutsideY,
+        tickLowerFeeGrowthOutsideX: { v: lowerTick.feeGrowthOutsideX },
+        tickLowerFeeGrowthOutsideY: { v: lowerTick.feeGrowthOutsideY },
         tickUpperIndex: upperTick.index,
-        tickUpperFeeGrowthOutsideX: upperTick.feeGrowthOutsideX,
-        tickUpperFeeGrowthOutsideY: upperTick.feeGrowthOutsideY,
+        tickUpperFeeGrowthOutsideX: { v: upperTick.feeGrowthOutsideX },
+        tickUpperFeeGrowthOutsideY: { v: upperTick.feeGrowthOutsideY },
         tickCurrent: pool.currentTickIndex,
-        globalFeeGrowthX: pool.feeGrowthGlobalX,
-        globalFeeGrowthY: pool.feeGrowthGlobalY,
-        positionFeeGrowthInsideX: position.feeGrowthInsideX,
-        positionFeeGrowthInsideY: position.feeGrowthInsideY,
-        positionLiquidity: position.liquidity
+        globalFeeGrowthX: { v: pool.feeGrowthGlobalX },
+        globalFeeGrowthY: { v: pool.feeGrowthGlobalY },
+        positionFeeGrowthInsideX: { v: position.feeGrowthInsideX },
+        positionFeeGrowthInsideY: { v: position.feeGrowthInsideY },
+        positionLiquidity: { v: position.liquidity }
       }
     })
   ).returns
+
+  return [returns[0].v, returns[1].v]
 }
 
 export const calculateTokenAmounts = async (
   pool: Pool,
   position: Position
 ): Promise<[bigint, bigint, boolean]> => {
-  return (
+  const returns = (
```

Comment:
To make it looking clearer what the function returns you can use array destructuring.